homeassistant/components/logbook.py

```diff
@@ -309,23 +298,41 @@ def _exclude_events(events, config):
         domain, entity_id = None, None
 
         if event.event_type == EVENT_STATE_CHANGED:
-            to_state = State.from_dict(event.data.get('new_state'))
+            entity_id = event.data.get('entity_id')
+
+            if entity_id is None:
+                continue
+
             # Do not report on new entities
             if event.data.get('old_state') is None:
                 continue
 
+            new_state = event.data.get('new_state')
+
             # Do not report on entity removal
-            if not to_state:
+            if not new_state:
+                continue
+
+            attributes = new_state.get('attributes', {})
+
+            # If last_changed != last_updated only attributes have changed
+            # we do not report on that yet.
+            last_changed = new_state.get('last_changed')
+            last_updated = new_state.get('last_updated')
+            if last_changed != last_updated:
+                continue
+
+            domain = split_entity_id(entity_id)[0]
+
+            # Also filter auto groups.
+            if domain == 'group' and attributes.get('auto', False):
                 continue
 
             # exclude entities which are customized hidden
-            hidden = to_state.attributes.get(ATTR_HIDDEN, False)
+            hidden = attributes.get(ATTR_HIDDEN, False)
             if hidden:
                 continue
 
-            domain = to_state.domain
-            entity_id = to_state.entity_id
-
         elif event.event_type == EVENT_LOGBOOK_ENTRY:
             domain = event.data.get(ATTR_DOMAIN)
             entity_id = event.data.get(ATTR_ENTITY_ID)
```

Comment:
We should migrate the filters below to `helpers/entity_filter.py`, as that one returns an optimized function for the passed in configuration.

Reply:
All the `excluded_domains` etc. filters? Agreed, though that seems like a separate opportunity that could fit in a separate PR :)